src/d3a/d3a_core/export.py

```diff
@@ -386,6 +389,56 @@ def _plot_ess_soc_history(self, storage_list: list, subdir: str, root_name: str)
         output_file = os.path.join(plot_dir, 'ess_soc_history_{}.html'.format(root_name))
         BarGraph.plot_bar_graph(barmode, title, xtitle, ytitle, data, output_file)
 
+    def supply_demand_curve(self, area, subdir):
+        """
+        Wrapper for _supply_demand_curve
+        """
+        new_subdir = os.path.join(subdir, area.slug)
+        self._supply_demand_curve(new_subdir, area)
+        for child in area.children:
+            if child.children:
+                self.supply_demand_curve(child, new_subdir)
+
+    def _supply_demand_curve(self, subdir: str, area: Area):
+
+        for i in area.past_markets:
+            data = list()
+            xmax = 0
+            for j, k in i.cumulative_offers.items():
+                data.append(self._line_plot(k, j, True))
+            for l, m in i.cumulative_bids.items():
+                data.append(self._line_plot(m, l, False))
+            for n, o in i.clearing.items():
```

Comment:
As stated in other comments, at least the 'o' parameter should be expanded to different names (it is a tuple, right?), because it is even more difficult to understand what this variable does if you have to keep in mind what is [0] and [1] in it.